components/Checkbox/src/index.scss

```diff
@@ -1,108 +1,118 @@
 .denhaag-checkbox {
-  border: 0;
-  cursor: pointer;
   display: inline-flex;
-  outline: 0;
-  padding-inline-start: 0;
-  padding-inline-end: 0;
-  padding-block-start: 0;
-  padding-block-end: 0;
-  position: relative;
-  align-items: center;
-  user-select: none;
-  border-radius: 0;
-  vertical-align: middle;
-  -moz-appearance: none;
   justify-content: center;
-  text-decoration: none;
-  background-color: transparent;
-  -webkit-appearance: none;
-  -webkit-tap-highlight-color: transparent;
-  width: 24px;
-  height: 24px;
-  margin-inline-start: var(--denhaag-checkbox-margin-inline-start);
-  margin-inline-end: var(--denhaag-checkbox-margin-inline-end);
-  margin-block-start: var(--denhaag-checkbox-margin-block-start);
-  margin-block-end: var(--denhaag-checkbox-margin-block-end);
-  color: var(--denhaag-checkbox-normal-unchecked-color);
-}
-
-.denhaag-checkbox:focus-within::after,
-.denhaag-checkbox--focus::after {
-  content: "";
-  display: block;
+  align-items: center;
+  position: relative;
+}
+
+.denhaag-checkbox__input {
+  z-index: 1;
   position: absolute;
-  top: -2px;
-  bottom: -2px;
-  left: -2px;
-  right: -2px;
-  border: var(--denhaag-focus-border);
-  border-radius: var(--denhaag-border-radius);
+  top: 0;
+  left: 0;
+  width: var(--utrecht-custom-checkbox-size);
+  margin-inline-start: 0;
+  margin-inline-end: 0;
+  margin-block-start: 0;
+  margin-block-end: 0;
+  aspect-ratio: 1;
+  opacity: 0%;
+  cursor: pointer;
 }
 
-.denhaag-checkbox--checked {
-  color: var(--denhaag-checkbox-normal-checked-color);
+.denhaag-checkbox__icon {
+  display: inline-flex;
+  justify-content: center;
+  align-items: center;
+  width: var(--utrecht-custom-checkbox-size);
+  aspect-ratio: 1;
+  background-color: var(--utrecht-custom-checkbox-background-color);
+  border-width: var(--utrecht-custom-checkbox-border-width);
+  border-style: solid;
+  border-color: var(--utrecht-custom-checkbox-border-color);
+  border-radius: var(--utrecht-custom-checkbox-border-radius);
```

Comment:
I think the `utrecht-form-input` border is identical. Maybe it's useful to add those tokens as fallback values so other organisations won't have to define as many tokens.